### DISCUSSION TITLE
希望使用pjax方式加载页面

### DISCUSSION THREAD

**Discussion by youyou-sudo** [libsgh/PanIndex#337] [Ideas]
最近了解到pjax方式来局部加载网页会很大程序上节省服务器的资源，局部加载网页也会很大程序上不重复加载JS和CSS，对于用户使用体验也很好，所以就问下

**Comment by libsgh** [comment]
最近在计划用vue-router应该和pjax差不多都是通过history去实现
这种方式会和PanIndex的url设计准则冲突，PanIndex的请求路径，虚拟了文件的目录结构
例如: http://localhost:5238/aa/bb/cc.txt， 对应文件（夹）路径是/aa/bb/cc.txt，要想在上面两种方案中保持原有的路由规则是比较困难的，pjax更是增加了后端的代码逻辑：后端渲染部分html到前端，而且SEO好像也不友好，这样的修改也是大刀阔斧，我还得好好想一想

**Reply by youyou-sudo** [reply]
> 最近在计划用vue-router应该和pjax差不多都是通过history去实现 这种方式会和PanIndex的url设计准则冲突，PanIndex的请求路径，虚拟了文件的目录结构 例如: [http://localhost:5238/aa/bb/cc.txt，](http://localhost:5238/aa/bb/cc.txt%EF%BC%8C) 对应文件（夹）路径是/aa/bb/cc.txt，要想在上面两种方案中保持原有的路由规则是比较困难的，pjax更是增加了后端的代码逻辑：后端渲染部分html到前端，而且SEO好像也不友好，这样的修改也是大刀阔斧，我还得好好想一想

OK，了解了，这个需求倒不急，可以鸽一搁，只是看到大部分网站都是这样风格，就提一下
